framework/test/unit/processing/spark-runtime-containers.test.ts

```diff
@@ -940,3 +942,176 @@ describe('With provided EKS cluster, the construct ', () => {
+    kubectlLambdaLayer: kubectlLayer,
+    defaultNodes: false,
+  });
+
+  const virtualCluster = emrEksCluster.addEmrVirtualCluster(emrEksClusterStack, {
+    name: 'test',
+  });
+
+  const policy = new ManagedPolicy(emrEksClusterStack, 'testPolicy', {
+    document: new PolicyDocument({
+      statements: [
+        new PolicyStatement({
+          resources: ['arn:aws:s3:::aws-data-analytics-workshop'],
+          actions: ['s3:GetObject'],
+        }),
+      ],
+    }),
+  });
+
+  const execRole = emrEksCluster.createExecutionRole(emrEksClusterStack, 'test', policy, 'nons', 'myExecRole');
+
+  emrEksCluster.addInteractiveEndpoint(emrEksClusterStack, 'AddInteractiveEndpoint', {
+    managedEndpointName: 'test',
+    executionRole: execRole,
+    virtualClusterId: virtualCluster.attrId,
+  });
+
+  const template = Template.fromStack(emrEksClusterStack);
+
+  test('should create an interactive endpoint with provided name and no configOverride', () => {
+    template.hasResourceProperties('Custom::EmrEksInteractiveEndpoint', {
+      endpointName: 'test',
+      configurationOverrides: Match.absent(),
+    });
+  });
+
+});
+
+
+describe('Test for grant job execution', () => {
+
+  const emrEksClusterStack = new Stack();
+
+  const kubectlLayer = new KubectlV27Layer(emrEksClusterStack, 'kubectlLayer');
+
+  const adminRole = Role.fromRoleArn(emrEksClusterStack, 'AdminRole', 'arn:aws:iam::123445678901:role/eks-admin');
+
+  const emrEksCluster = SparkEmrContainersRuntime.getOrCreate(emrEksClusterStack, {
+    eksAdminRole: adminRole,
+    publicAccessCIDRs: ['10.0.0.0/32'],
+    kubectlLambdaLayer: kubectlLayer,
+  });
+
+  const virtualCluster = emrEksCluster.addEmrVirtualCluster(emrEksClusterStack, {
+    name: 'test',
+  });
+
+  const policy = new ManagedPolicy(emrEksClusterStack, 'testPolicy', {
+    document: new PolicyDocument({
+      statements: [
+        new PolicyStatement({
+          resources: ['arn:aws:s3:::aws-data-analytics-workshop'],
+          actions: ['s3:GetObject'],
+        }),
+      ],
+    }),
+  });
+
+  const execRole = emrEksCluster.createExecutionRole(emrEksClusterStack, 'test', policy, 'nons', 'myExecRole');
+
+  const startJobRole: IRole = Role.fromRoleName(emrEksClusterStack, 'StartJobRole', 'StartJobRole');
+
+  SparkEmrContainersRuntime.grantStartJobExecution(startJobRole, [execRole.roleArn], virtualCluster.attrArn);
+
+  const template = Template.fromStack(emrEksClusterStack);
+
+  test('should have a role with a policy to start a job in emr on eks', () => {
+    template.hasResourceProperties('AWS::IAM::Policy',
+      Match.objectLike({
+        PolicyDocument: Match.objectLike({
+          Statement: Match.arrayWith([
+            Match.objectLike(
+              {
+                Action: 'emr-containers:StartJobRun',
+                Effect: 'Allow',
+              },
+            ),
+          ]),
+        }),
+        Roles: ['StartJobRole'],
+      }));
+  });
+
+
+});
```

Comment:
can you add tests for the RemovalPolicy?  Something similar to this:

```
  test(' create resources with Delete policy', () => {
    template.allResources('AWS::Logs::LogGroup', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });

    template.allResources('Custom::RedshiftDataSql', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });

    template.allResources('Custom::LambdaEniCleanup', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });
  });
```

And same thing for the revert. This is to ensure all resources are deleted properly.